sys/linux/init.go

```diff
@@ -223,6 +231,21 @@ func (arch *arch) neutralize(c *prog.Call) {
 			family.Val = ^uint64(0)
 		}
 	case "syz_open_dev":
+		dev_ptr := c.Args[0].(*prog.PointerArg)
+		if dev_ptr.Res == nil {
+			// TODO: what to do here?
+			return
+		}
+
+		flags := c.Args[2].(*prog.ConstArg)
+
+		switch v := dev_ptr.Res.(type) {
+		case *prog.ConstArg:
+			if string(v.Val) == string("/dev/cpu/#/msr") && uint64(uint32(flags.Val)) == arch.O_RDWR {
+				flags.Val = arch.O_RDONLY
```

Comment:
That `enforceIntArg(c.Args[2])` below, won't it do this already?
Please add a test to sys/linux/init.go and it will answer the question.

Reply:
I didn't know. [This commit](https://github.com/google/syzkaller/commit/dfe2e9d84a64066fd15913c7cd02d1853adf3942) clarified things. Thank you